File: frontend/src/types/maintenance.ts
export enum RequestStatus {
    NEW = 'new',
    IN_PROGRESS = 'in_progress',
    REPAIRED = 'repaired',
    SCRAP = 'scrap'
}

export enum RequestCategory {
    CORRECTIVE = 'corrective',
    PREVENTIVE = 'preventive'
}

export enum RequestPriority {
    LOW = 'low',
    MEDIUM = 'medium',
    HIGH = 'high',
    URGENT = 'urgent'
}

export interface MaintenanceRequest {
    id: number;
    subject: string;
    description: string | null;
    category: string;
    status: string;
    priority: string;
    equipment_id: number;
    requester_id: number;
    technician_id: number | null;
    maintenance_team_id?: number | null;
    scheduled_date: string | null;
    completion_date: string | null;
    duration: number | null; // Duration in hours
    created_at: string;
    updated_at: string;
    company: string;
}

export interface MaintenanceRequestCreate {
    subject: string;
    description?: string;
    category: RequestCategory;
    priority: RequestPriority;
    equipment_id: number;
    technician_id?: number | null;
    maintenance_team_id?: number | null;
    scheduled_date?: string | null;
    duration: number | null; // Duration in hours
}

export interface MaintenanceRequestUpdate {
    subject?: string;
    description?: string;
    category?: RequestCategory;
    status?: RequestStatus;
    priority?: RequestPriority;
    technician_id?: number | null;
    maintenance_team_id?: number | null;
    scheduled_date?: string | null;
    completion_date?: string | null;
}

File: backend/app/schemas/maintenance.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional
from ..models.maintenance_request import RequestCategory, RequestStatus, RequestPriority


class MaintenanceRequestBase(BaseModel):
    subject: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    category: RequestCategory
    priority: RequestPriority
    equipment_id: int
    technician_id: Optional[int] = None
    maintenance_team_id: Optional[int] = None
    scheduled_date: Optional[str] = None  # Accept as string from frontend
    duration: Optional[float] = None  # Duration in hours


class MaintenanceRequestCreate(MaintenanceRequestBase):
    """Schema for creating a maintenance request"""
    pass


class MaintenanceRequestUpdate(BaseModel):
    """Schema for updating a maintenance request - all fields optional"""
    subject: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    category: Optional[RequestCategory] = None
    status: Optional[RequestStatus] = None
    priority: Optional[RequestPriority] = None
    priority: Optional[RequestPriority] = None
    technician_id: Optional[int] = None
    maintenance_team_id: Optional[int] = None
    scheduled_date: Optional[datetime] = None
    completion_date: Optional[datetime] = None


class MaintenanceRequestResponse(BaseModel):
    """Schema for maintenance request response"""
    id: int
    subject: str
    description: Optional[str] = None
    category: RequestCategory
    priority: RequestPriority
    equipment_id: int
    requester_id: int
    requester_id: int
    technician_id: Optional[int] = None
    maintenance_team_id: Optional[int] = None
    status: RequestStatus
    scheduled_date: Optional[datetime] = None
    completion_date: Optional[datetime] = None
    duration: Optional[float] = None
    created_at: datetime
    updated_at: datetime
    company: str

    class Config:
        from_attributes = True
